File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/ReadPropertyHistoryValuesResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.DataValue;

import java.util.ArrayList;
import java.util.List;

public class ReadPropertyHistoryValuesResponse {
    List<DataValue> dataValues;
    boolean containsMoreData;

    public ReadPropertyHistoryValuesResponse()
    {
        dataValues = new ArrayList<>();
        containsMoreData = false;
    }

    public ReadPropertyHistoryValuesResponse(
            List<DataValue> dataValues,
            boolean containsMoreData)
    {
        this.dataValues = dataValues;
        this.containsMoreData = containsMoreData;
    }

    public void addDataValue(DataValue value)
    {
        dataValues.add(value);
    }

    public List<DataValue> getDataValues() {
        return dataValues;
    }

    public void setContainsMoreData(boolean containsMoreData) {
        this.containsMoreData = containsMoreData;
    }

    public boolean containsMoreData() {
        return containsMoreData;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/ReadPropertyHistoryValuesResponse.ts
import { UaDataValue } from "opcua-webapi-ts";

export class ReadPropertyHistoryValuesResponse {
    private readonly _dataValues: Array<UaDataValue>;
    private _containsMoreData: boolean;

    constructor() {
        this._dataValues = [];
        this._containsMoreData = false;
    }

    addDataValue(value: UaDataValue): void {
        this._dataValues.push(value);
    }

    get dataValues(): Array<UaDataValue> {
        return this._dataValues;
    }

    set containsMoreData(containsMoreData: boolean) {
        this._containsMoreData = containsMoreData;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }
}